services/terrain/src/terrain/services/permanent_id_requests.clj

```diff
@@ -0,0 +1,149 @@
+(ns terrain.services.permanent-id-requests
+  (:use [kameleon.uuids :only [uuidify]]
+        [slingshot.slingshot :only [throw+]]
+        [terrain.auth.user-attributes :only [current-user]])
+  (:require [cheshire.core :as json]
+            [clojure.tools.logging :as log]
+            [clojure-commons.file-utils :as ft]
+            [terrain.clients.data-info :as data-info]
+            [terrain.clients.data-info.raw :as data-info-client]
+            [terrain.clients.metadata.raw :as metadata]
+            [terrain.util.config :as config]
+            [terrain.util.service :as service]))
+
+(defn- validate-request-target-type
+  [{target-type :type :as folder}]
+  (let [target-type (metadata/resolve-data-type target-type)]
+    (when-not (= target-type "folder")
+      (throw+ {:type :clojure-commons.exception/bad-request-field
+               :error "The given data ID does not belong to a folder."
+               :file folder}))
+    target-type))
+
+(defn- validate-owner
+  [user {:keys [id path] :as data-item}]
+  (when-not (data-info/owns? user path)
+    (throw+ {:type :clojure-commons.exception/not-owner
+             :error "User does not own given folder."
+             :user user
+             :folder-id id}))
+  data-item)
+
+(defn- validate-staging-dest
+  [{:keys [path] :as data-item}]
+  (let [staging-dest (ft/path-join (config/permanent-id-staging-dir) (ft/basename path))]
+    (when (data-info/path-exists? (config/irods-user) staging-dest)
+    (throw+ {:type :clojure-commons.exception/exists
+             :error "A folder with this name has already been submitted for a Permanent ID request."
+             :path staging-dest})))
+  data-item)
+
+(defn- validate-data-item
+  [user data-item]
+  (->> data-item
+       (validate-owner user)
+       validate-staging-dest))
+
+(defn- get-requested-data-item
+  "Gets data-info stat for the given ID and checks if the data item is valid for a Permanent ID request."
+  [user data-id]
+  (->> data-id
+       (data-info/stat-by-uuid user)
+       (validate-data-item user)))
```

Comment:
(originally saw this on the other PR but it got added here)

It may make sense to specifically pass the validation behavior to data-info while getting the stat information, rather than using the owner (understanding, of course, that I added that to the endpoint while you were writing this, so it's quite reasonable you didn't), to reduce the number of data-info requests that need to be made.